File: Baekjoon/Class4/09.1149 RGB거리.py
import sys
input = sys.stdin.readline

n = int(input())
r, g, b = map(int, input().split())

for i in range(1, n):
    prev_r, prev_g, prev_b = r, g, b

    r, g, b = map(int, input().split())
    r += min(prev_g, prev_b) #다른 색 중 최솟값 더하기
    g += min(prev_r, prev_b)
    b += min(prev_r, prev_g)

print(min(r, g, b))

''' 확장 가능
import sys
input = sys.stdin.readline

MAX_COST = 10000000

n = int(input())
colors = list(map(int, input().split()))

for i in range(1, n):
    prev_colors = colors

    colors = list(map(int, input().split()))
    colors = [colors[j] + min([
        #다른 색 중 최솟값 더하기
        prev_colors[k] if j != k else MAX_COST for k in range(0, len(colors))
        ]) for j in range(len(colors))]

print(min(colors))
'''

File: Baekjoon/Class4/09.1149 RGB거리.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const n = Number(input[0]);
let [r, g, b] = input[1].split(" ").map(Number);

for(let i = 2; i <= n; i++) {
    const prevR = r, prevG = g, prevB = b;
    
    [r, g, b] = input[i].split(" ").map(Number);
    r += Math.min(prevG, prevB); // 다른 색 중 최솟값 더하기
    g += Math.min(prevR, prevB);
    b += Math.min(prevR, prevG);
}

console.log(Math.min(r, g, b));

/* 확장 가능
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const MAX_COST = 10000000;

const n = Number(input[0]);
let colors = input[1].split(" ").map(Number);

for(let i = 2; i <= n; i++) {
    const prevColors = colors;
    
    colors = input[i].split(" ").map(Number);
    for(let j = 0; j < colors.length; j++) { // 다른 색 중 최솟값 더하기
        let minPrevColor = MAX_COST;
        for(let k = 0; k < colors.length; k++)
            if(j !== k)
                minPrevColor = Math.min(minPrevColor, prevColors[k])
        colors[j] += minPrevColor;
    }
}

console.log(Math.min(...colors));
*/